app/src/main/java/in/ivlabs/shayak/view/MainActivity.kt

```diff
@@ -1,12 +1,101 @@
 package `in`.ivlabs.shayak.view
 
+import `in`.ivlabs.shayak.mainactivity.MainActivityPresenterInterface
+import `in`.ivlabs.shayak.mainactivity.MainActivityViewInterface
+import android.graphics.Color
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.ivlabs.shayak.R
+import com.mikepenz.fastadapter.FastAdapter
+import com.mikepenz.fastadapter.adapters.ItemAdapter
+import com.mikepenz.fastadapter.items.AbstractItem
 
-class MainActivity : AppCompatActivity() {
+val Boolean.int
+    get() = if (this) 1 else 0
+
+class MainActivity : AppCompatActivity(), MainActivityViewInterface,
+    MainActivityPresenterInterface {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        displayRobotList(getDummyRobotList())
+    }
+
+    override fun displayRobotList(list: List<MainActivityViewInterface.RobotViewData>) {
+        val robotAdapter = ItemAdapter<RobotItem>()
+        val fastAdapter = FastAdapter.with(robotAdapter)
+        findViewById<RecyclerView>(R.id.main_recycler).adapter = fastAdapter
+        findViewById<RecyclerView>(R.id.main_recycler).layoutManager =
+            LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+        robotAdapter.add(
+            list.sortedWith(Comparator { o1: MainActivityViewInterface.RobotViewData, o2: MainActivityViewInterface.RobotViewData -> o2.isFavorite.int - o1.isFavorite.int })
+                .map {
+                    RobotItem(it)
+                }.toList()
+        )
+        Toast.makeText(this, "Here!", Toast.LENGTH_SHORT).show()
+    }
+
+    override fun robotConnected(UUID: String) {
+        TODO("Not yet implemented")
     }
+
+    override fun connectToRobot(UUID: String): Boolean {
+        TODO("Not yet implemented")
+    }
+
+    /*Gets Dummy List of Robots to test RecyclerView Population*/
+    fun getDummyRobotList(): List<MainActivityViewInterface.RobotViewData> {
+        val ROBOTS_TO_DISPLAY = List<MainActivityViewInterface.RobotViewData>(5) {
+            MainActivityViewInterface.RobotViewData(
+                "$it",
+                "Robot $it",
+                "Short Description of Robot $it",
+                it % 2 == 0
+            )
+        }
+        return ROBOTS_TO_DISPLAY
+    }
+}
+
+open class RobotItem(robot: MainActivityViewInterface.RobotViewData) :
```

Comment:
Why is this an open class? We dont need to extend this class mark it as closed
```suggestion
class RobotItem(robot: MainActivityViewInterface.RobotViewData) :
```